packages/core/src/connectors/index.ts

```diff
@@ -12,13 +19,22 @@ import * as WeChat from './wechat';
 const allConnectors: IConnector[] = [AliyunDM, AliyunSMS, Facebook, GitHub, Google, WeChat];
 
 export const getConnectorInstances = async (): Promise<ConnectorInstance[]> => {
-  return Promise.all(
-    allConnectors.map(async (element) => {
-      const connector = await findConnectorById(element.metadata.id);
+  const connectors = await findAllConnectors();
+  const connectorMap = new Map(connectors.map((connector) => [connector.id, connector]));
+  const connectorInstances = allConnectors
+    .map((element) => {
+      const { id } = element.metadata;
+      const connector = connectorMap.get(id);
+
+      if (!connector) {
+        throw new RequestError({ code: 'entity.not_found', id, status: 404 });
+      }
 
       return { connector, ...element };
     })
-  );
+    .filter((entry): entry is ConnectorInstance => notFalsy(entry));
```

Comment:
Looks like, the entry won't be falsy? as you throws the error out?

Reply:
It's used to make entry as `ConnectorInstance` type.

You can have a try, and maybe you can find a better way to do it?